week2/week2_HW/Program.cs

```diff
@@ -0,0 +1,139 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+//Exercise1 Calculator
+
+app.MapGet("/calculator", (string order, string num1, string num2) =>
+
+{
+    var outPut = 0;
+
+    var numberOne = int.TryParse(num1, out var numberOneParsed);
+    var numberTwo = int.TryParse(num2, out var numberTwoParsed);
+
+    if (!numberTwo || !numberOne)
+    {
+        return Results.Ok(new String("One of the number is not validated"));
+    }
+
+    switch (order)
+    {
+        case "add":
+            outPut = numberOneParsed + numberTwoParsed;
+            break;
+        case "subtract":
+            outPut = numberOneParsed - numberTwoParsed;
+            break;
+        case "multiplication":
+            outPut = numberOneParsed * numberTwoParsed;
+            break;
+        default:
+           return  Results.Ok(new String($"requested order is not supported"));
+
+
+    };
+
+    
+    return Results.Ok(new String($"Answer is {outPut}"));
+
+});
+
+
+//Exercise2  count capital letters or assumption of numbers
+
+
+app.MapGet("/detector", (string input) => {
+
+    int result;
+
+    var parsedInput = int.TryParse(input, out var inputAsNumber);
+
+
+     if (!parsedInput)
+     {
+         result=CountCapitalLetters(input);
+     }
+      else
+     {  
+         result = AddNumbers(input); 
+     }
+
+    return $"your answer is {result}";
+
+});
+
+
+
+int AddNumbers (string input) 
+{
+
+    int sumOfAll = 0;
+    foreach (var item in input )
+    {
+    
+        int numberedChar = int.Parse(item.ToString());
+        sumOfAll += numberedChar ;
+    }
+  
+    return sumOfAll;
+}
+
+int CountCapitalLetters(string input ){
+
+    int sumOfCapitalChars = 0;
+
+    for (int i = 0; i < input.Length; i++)
+    {
+        if (char.IsLetter(input[i])&& char.IsUpper(input[i]))
+        {
+            sumOfCapitalChars++;
+        }
+    }
+    return sumOfCapitalChars;
+}
+
+
+//Exercise3 Sort Characters
+
+app.MapGet("/alphabetical", (string input) => {
+
+    var separatedChars = new List<char>();
+
+    for (int i = 0; i < input.Length; i++)
+    { 
+        if (char.IsLetter(input[i]))

```

Comment:
You also need to check if a character is not already part of the list. You can do it by calling 
```suggestion
        if (char.IsLetter(input[i]) && !separatedChars.Contains(input[i]))
```

And another way, that we don't expect students to do, is by using [distinct](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.distinct?view=net-7.0) & `OrderBy` extension method. Here we could apply expression body since it would be a one-liner.
```
app.MapGet("/alphabetical", (string input) => input.Distinct().ToList().OrderBy(s => s);
```